### ISSUE TITLE
Playlist playback stops after a song

### ISSUE THREAD

**Issue by kyokusa** [KoljaWindeler/ytube_music_player#157]
Hi,
any playlist I start, after a song it stops.
This happened after the last update to:
Home Assistant 2021.12.0b4    -    ytube_music_player-20211209.01

My configuration:
- type: custom:mini-media-player
     entity: media_player.ytube_music_player
     artwork: cover
     shortcuts:
     columns: 6
     buttons:
        - name: a-Ha          
          type: Playlist
          id: PLUxwfwf3SeRxsOiTpepSCvUSRF-xxxxxxxx

I hope for some solution and thanks.

**Comment by KoljaWindeler** [comment]
Please enable debug output (see Readme) and post the log around the time the first song finishes

**Comment by kyokusa** [comment]
here it is thanks:

2021-12-10 05:19:09 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration localtuya which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-12-10 05:19:09 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration scheduler which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-12-10 05:19:09 WARNING (SyncWorker_3) [homeassistant.loader] We found a custom integration daikin_residential which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-12-10 05:19:09 WARNING (SyncWorker_4) [homeassistant.loader] We found a custom integration multizone_controller which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-12-10 05:19:09 WARNING (SyncWorker_2) [homeassistant.loader] We found a custom integration sonoff which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-12-10 05:19:09 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-12-10 05:19:09 WARNING (SyncWorker_4) [homeassistant.loader] We found a custom integration tuya_local which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-12-10 05:19:09 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration ytube_music_player which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant

**Comment by KoljaWindeler** [comment]
Hi, thanks. But thats not what I'm looking for. 
That's the part right after the start ... Please have a look at the timestamps.
There will be more log output further down ..

**Comment by kyokusa** [comment]
Hi, sorry but I don't know how to get homeAssistant debugging.
I added this to configuration.yaml:

logger:
    default: error
    logs:
      homeassistant.components.sensor.deluge: debug

the log in the ¨hassio / homeassistant / home-assistant.log¨ folder is the same as that of the program

**Comment by KoljaWindeler** [comment]
Please read the readme .. https://github.com/KoljaWindeler/ytube_music_player#readme

It has a section on how to enable debug .. no need for any configurations

**Comment by kyokusa** [comment]
OK Thanks:

2021-12-10 21:36:50 INFO (MainThread) [custom_components.hacs] Stage changed: HacsStage.RUNNING
2021-12-10 21:36:55 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] play_media, media_type: playlist, media_id: PLUp6Lpo17xsOiTpepSCvUSRF-l3aKe85c
2021-12-10 21:36:55 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_prepare_play
2021-12-10 21:36:55 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_check_api [E]
2021-12-10 21:36:55 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_update_remote_player(Input / current media_player.nest_hub_mansarda) [E]
2021-12-10 21:36:55 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_turn_on_media_player [E]
2021-12-10 21:36:55 INFO (SyncWorker_1) [pychromecast] Quiting current app
2021-12-10 21:36:55 INFO (SyncWorker_1) [pychromecast.controllers] Receiver:Stopping current app 'E8C28D3C'
2021-12-10 21:36:55 INFO (SyncWorker_1) [pychromecast.controllers] Receiver:Launching app B45F4572
2021-12-10 21:36:56 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_get_cipher [E]
2021-12-10 21:36:56 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [E] async_prepare_play
2021-12-10 21:36:57 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] get_playlist(playlistId=PLUp6Lpo17xsOiTpepSCvUSRF-l3aKe85c)
2021-12-10 21:36:57 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] - shuffle new tracklist
2021-12-10 21:36:57 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] _tracks_to_attribute
2021-12-10 21:36:57 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_update_extra_sensor [E]
2021-12-10 21:36:57 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_update_extra_sensor [E]
2021-12-10 21:36:57 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [E] _tracks_to_attribute
2021-12-10 21:36:57 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] _play
2021-12-10 21:36:57 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_get_track
2021-12-10 21:36:57 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] - Playing track nr 1 / 100
2021-12-10 21:36:57 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_get_url
2021-12-10 21:36:57 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_check_api [E]
2021-12-10 21:36:57 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] - try to find URL on our own
2021-12-10 21:36:57 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_sync_player ... media_player.nest_hub_mansarda: off -> playing ... not playing [E]
2021-12-10 21:36:57 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_sync_player ... media_player.nest_hub_mansarda: playing -> playing ... not playing [E]
2021-12-10 21:36:57 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] - found URL in api data
2021-12-10 21:36:57 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [E] async_get_url
2021-12-10 21:36:57 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] - forwarding url to player media_player.nest_hub_mansarda
2021-12-10 21:36:57 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_update_extra_sensor [E]
2021-12-10 21:36:57 INFO (SyncWorker_5) [pychromecast.controllers] Not launching app B45F4572 - already running
2021-12-10 21:36:57 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_sync_player ... media_player.nest_hub_mansarda: playing -> playing
2021-12-10 21:36:57 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] detected different app _id, shuttiung down without interruption
2021-12-10 21:36:57 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_turn_off_media_player [E]
2021-12-10 21:36:58 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [E] async_get_track
2021-12-10 21:36:58 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [E] play_media

**Comment by KoljaWindeler** [comment]
Hm interesting .. I've always expected that this could be a problem. 
Will work an on solution and get back to you. 

Out of curiosity: what device are you using?

**Comment by KoljaWindeler** [comment]
And a nest hub as playback device?

**Comment by kyokusa** [comment]
Yes Version 1

**Comment by kyokusa** [comment]
If you find it useful, I'll make you a short video with the dynamics of the error

**Comment by KoljaWindeler** [comment]
No need, thanks. The issue is clear

**Comment by KoljaWindeler** [comment]
Please test the latest released beta version 20211212

**Comment by kyokusa** [comment]
OK, I did the update to version 20211212, it looks ok, but I noticed that after a certain time the Google Nest Hub goes back to slide show mode, even if the playback of the songs still continues. Great job thanks.

**Comment by rickyxsosa** [comment]
Hello,

This may need to be reopened, not sure if something else changed

I am getting the same issue now 

ytube version: 20211213.01
home assistant core version: core-2021.12.3


here is the script I use to start it

```
alias: Kids Bedtime
sequence:
  - scene: scene.kids_bedtime_scene
  - service: media_player.select_source
    target:
      entity_id: media_player.ytube_music_player
    data:
      source: media_player.kids_speaker
  - service: media_player.play_media
    target:
      entity_id: media_player.ytube_music_player
    data:
      media_content_id: PLlm0jT8rxW3YiZefj11xxxxxxxxxxxxxx
      media_content_type: playlist
mode: single
```


and here is the debug entries

```
2021-12-20 09:50:38 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_select_source(media_player.kids_speaker)
2021-12-20 09:50:38 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_update_remote_player(Input media_player.kids_speaker/ current media_player.kids_speaker) 
2021-12-20 09:50:38 DEBUG (MainThread) [custom_components.ytube_music_player.media_player]  no change [E]
2021-12-20 09:50:38 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] - Choosing media_player.kids_speaker as player
2021-12-20 09:50:38 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [E] async_select_source
2021-12-20 09:50:38 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] play_media, media_type: playlist, media_id: PLlm0jT8rxW3YiZefj11xxxxxxxxxxxxxx
2021-12-20 09:50:38 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_prepare_play
2021-12-20 09:50:38 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_check_api [E]
2021-12-20 09:50:38 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_update_remote_player(Input / current media_player.kids_speaker)  [E]
2021-12-20 09:50:38 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_turn_on_media_player [E]
2021-12-20 09:50:39 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_get_cipher [E]
2021-12-20 09:50:39 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [E] async_prepare_play
2021-12-20 09:50:39 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] get_playlist(playlistId=PLlm0jT8rxW3YiZefj11xxxxxxxxxxxxxx)
2021-12-20 09:50:39 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] - shuffle new tracklist
2021-12-20 09:50:39 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] _tracks_to_attribute
2021-12-20 09:50:39 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_update_extra_sensor [E]
2021-12-20 09:50:39 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_update_extra_sensor [E]
2021-12-20 09:50:39 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [E] _tracks_to_attribute
2021-12-20 09:50:39 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] _play
2021-12-20 09:50:39 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_get_track
2021-12-20 09:50:39 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] - Playing track nr 1 / 20
2021-12-20 09:50:39 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_get_url
2021-12-20 09:50:39 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_check_api [E]
2021-12-20 09:50:39 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] - try to find URL on our own
2021-12-20 09:50:40 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] - self decoded URL via cipher
2021-12-20 09:50:40 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [E] async_get_url
2021-12-20 09:50:40 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] - forwarding url to player media_player.kids_speaker
2021-12-20 09:50:40 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_update_extra_sensor [E]
2021-12-20 09:50:41 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [E] async_get_track
2021-12-20 09:50:41 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [E] play_media
2021-12-20 09:50:44 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_sync_player ... media_player.kids_speaker: off -> playing
2021-12-20 09:50:44 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] detected app _id, B45F4572
2021-12-20 09:50:44 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [E] async_sync_player
2021-12-20 09:50:56 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_sync_player ... media_player.kids_speaker [E]
2021-12-20 09:59:21 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_sync_player ... media_player.kids_speaker: playing -> off
2021-12-20 09:59:21 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] media player got turned off
2021-12-20 09:59:21 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] turn_off
2021-12-20 09:59:21 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [S] async_turn_off_media_player [E]
2021-12-20 09:59:21 DEBUG (MainThread) [custom_components.ytube_music_player.media_player] [E] async_sync_player
2021-12-20 09:59:21 WARNING (SyncWorker_1) [pychromecast.controllers] STOP command requested but no session is active.

```

**Comment by KoljaWindeler** [comment]
This is due to a bug in Homeassistent. There was a pull request. Let's hope that the next version works again

**Comment by stboch** [comment]
this bug has been resolved in released version 2021.12.4
